File: languages/go/record.go
package bloock

import (
	"log"

	"github.com/bloock/go-bloock/internal/ffi"
)

type Record struct {
	ffi *ffi.RecordFfi
}

func newRecord(f *ffi.RecordFfi) *Record {
	return &Record{
		ffi: f,
	}
}

func NewRecordFromHash(hash string) (*Record, error) {
	res, err := ffi.NewRecordFromHash(hash)
	if err != nil {
		return nil, err
	}
	return newRecord(res), nil
}

func NewRecordFromHex(hex string) (*Record, error) {
	res, err := ffi.NewRecordFromHex(hex)
	if err != nil {
		return nil, err
	}
	return newRecord(res), nil
}

func NewRecordFromString(str string) (*Record, error) {
	res, err := ffi.NewRecordFromString(str)
	if err != nil {
		return nil, err
	}
	return newRecord(res), nil
}

func (r *Record) GetHash() string {
	hash, err := r.ffi.GetHash()
	if err != nil {
		log.Printf("error retrieving record hash: %s", err.Error())
	}
	return hash
}

File: languages/js/src/record.ts
import { BloockBridge } from "./bridge/bridge";
import { FromHashRequest, FromHexRequest, FromStringRequest, FromTypedArrayRequest, Record as ProtoRecord } from "./bridge/proto/record"

export class Record {
    static FromHash(hash: string): Promise<ProtoRecord> {
        let request = FromHashRequest.fromPartial({
            hash: hash,
        });

        let bridge = new BloockBridge();

        return new Promise((resolve, reject) => {
            bridge.getRecord().fromHash(request, (err, res) => {
                if (err) {
                    reject(err);
                }

                resolve(res);
            });
        });
    }

    static FromHex(hex: string): Promise<ProtoRecord> {
        let request = FromHexRequest.fromPartial({
            hex: hex,
        });

        let bridge = new BloockBridge();

        return new Promise((resolve, reject) => {
            bridge.getRecord().fromHex(request, (err, res) => {
                if (err) {
                    reject(err);
                }

                if (res.error) {
                    reject(res.error.message);
                }

                resolve(res.record!);
            });
        });
    }

    static FromString(str: string): Promise<ProtoRecord> {
        let request = FromStringRequest.fromPartial({
            str: str
        });

        let bridge = new BloockBridge();

        return new Promise((resolve, reject) => {
            bridge.getRecord().fromString(request, (err, res) => {
                if (err) {
                    reject(err);
                }

                resolve(res);
            });
        });
    }

    static FromTypedArray(array: Buffer): Promise<ProtoRecord> {
        let request = FromTypedArrayRequest.fromPartial({
            array: array
        });

        let bridge = new BloockBridge();

        return new Promise((resolve, reject) => {
            bridge.getRecord().fromTypedArray(request, (err, res) => {
                if (err) {
                    reject(err);
                }

                resolve(res);
            });
        });
    }
}
